### PR TITLE
Setup for gatsby

### LINKED ISSUE #6: Are there changes that need to be made to the project structure for Gatsby plugin development?

### PR DESCRIPTION
**Pull request by ebinion**
Resolves #6

So it turns out that Gatsby wants all of the build files for the plugin in the root of the NPM module. I don't love this from the perspective of avoiding committing the build files to the repo. 
@mgreich any ideas?

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+.lock-wscript
+
+# Dependency directory
+# https://www.npmjs.org/doc/misc/npm-faq.html#should-i-check-my-node_modules-folder-into-git
+node_modules
+*.un~
+yarn.lock
+package-lock.json
+source
+.nvmrc
```
Comment (mgreich) [inline .npmignore]: Do we want to ignore this?

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+.hound.yml
+.eslintrc.yml
```
Comment (mgreich) [inline .npmignore]: Or this.

File: .npmignore [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+.eslintignore
```
Comment (mgreich) [inline .npmignore]: Or this.

**Review by mgreich — COMMENTED** [review]
Do we want to ignore the .nvmrc and eslint files? I guess I'm not sure how to handle those things. If you want others to contribute they'd be helpful, but maybe don't belong in an NPM package???

File: .npmignore (hunk 1 above)
Comment (ebinion) [reply .npmignore]: Yes. (Explanation coming)

File: .npmignore (hunk 3 above)
Comment (ebinion) [reply .npmignore]: I think so. Like you mentioned, this is the ignore file for distribution on NPM only. The end result is intended to be that the package file (required for modules) and built file (the compiled Gatsby plugin) are the only things shipped through NPM (mimicking the pattern of Gatsby's official plugins. 

Folks work on the Git repo will have everything they need contribute.

From my (admittedly limited) understanding of how NPM works, I think only one of us will be able to push things up to NPM publishing is done through an account.

**Pull request merged**